internal/pkg/describe/backend_service.go

```diff
@@ -79,67 +79,118 @@ func (d *BackendServiceDescriber) URI(envName string) (string, error) {
 	return s.String(), nil
 }
 
+type backendSvcStackInfo struct {
+	config           *ServiceConfig
+	serviceDiscovery serviceDiscovery
+	envVars          map[string]string
+	err              error
+}
+
+func (d *BackendServiceDescriber) describeStackInfo(env string) backendSvcStackInfo {
+	svcParams, err := d.svcDescriber[env].Params()
+	if err != nil {
+		return backendSvcStackInfo{err: fmt.Errorf("retrieve service deployment configuration: %w", err)}
+	}
+	config := &ServiceConfig{
+		Environment: env,
+		Port:        svcParams[stack.LBWebServiceContainerPortParamKey],
+		Tasks:       svcParams[stack.ServiceTaskCountParamKey],
+		CPU:         svcParams[stack.ServiceTaskCPUParamKey],
+		Memory:      svcParams[stack.ServiceTaskMemoryParamKey],
+	}
+	sd := serviceDiscovery{
+		Service: d.svc,
+		Port:    svcParams[stack.LBWebServiceContainerPortParamKey],
+		App:     d.app,
+		Env:     env,
+	}
+	backendSvcEnvVars, err := d.svcDescriber[env].EnvVars()
+	if err != nil {
+		return backendSvcStackInfo{err: fmt.Errorf("retrieve environment variables: %w", err)}
+	}
+	return backendSvcStackInfo{
+		config:           config,
+		envVars:          backendSvcEnvVars,
+		serviceDiscovery: sd,
+	}
+}
+
+func (d *BackendServiceDescriber) svcStackResources(envs []string) (map[string][]*CfnResource, error) {
+	resources := make(map[string][]*CfnResource)
+	if d.enableResources {
+		resourceCh := make(chan svcStackResources, len(envs))
+		defer close(resourceCh)
+		for _, env := range envs {
+			go func(env string) {
+				err := d.initServiceDescriber(env)
+				if err != nil {
+					resourceCh <- svcStackResources{err: err}
+					return
+				}
+				stackResources, err := d.svcDescriber[env].ServiceStackResources()
+				if err != nil {
+					resourceCh <- svcStackResources{err: fmt.Errorf("retrieve service resources: %w", err)}
+					return
+				}
+				resourceCh <- svcStackResources{resource: stackResources}
+			}(env)
+		}
+		for _, env := range envs {
+			res := <-resourceCh
+			if res.err != nil {
+				return nil, res.err
+			}
+			resources[env] = flattenResources(res.resource)
```

Comment:
Mmm the responses can come in any order now that it's concurrent. So it's not necessary that when env == "test" the response will be for "test", it can be for "prod" instead. We should add the environment name to the response instead.

Reply:
oh yeah that's a good catch